### PR TITLE
Add pj_phi2_test.

### PR DESCRIPTION
**Pull request by schwehr**
My first attempt at a GoogleTest based test in PROJ proper.

License intentionally changed from Apache 2 to match PROJ

Nigrated from:

https://github.com/schwehr/gdal-autotest2/blob/master/cpp/third_party/proj/pj_phi2_test.cc

### CODE REVIEW

**Comment by schwehr** [comment]
I expect this to take a few iterations to get correct

**Comment by rouault** [comment]
Please include it in test/unit/Makefile.am and test/unit/CMakeLists.txt

**Comment by schwehr** [comment]
Arg... using CLion for git I missed that it didn't add those

File: test/unit/Makefile.am [hunk 1]
```diff
@@ -7,12 +7,19 @@ include_HEADERS = gtest_include.h
...
 basic_test-check: basic_test
 	./basic_test
 
+pj_phi2_test_SOURCES = pj_phi2_test.cpp main.cpp
+pj_phi2_test_LDADD = ../../src/libproj.la ../../test/googletest/libgtest.la
+
+pj_phi2_test-check: pj_phi2_test
+	./pj_phi2_test
+
 check-local: basic_test-check
```
Comment (schwehr) [inline test/unit/Makefile.am:25]: What is check-local?

File: test/unit/pj_phi2_test.cpp [hunk 2]
```diff
@@ -0,0 +1,85 @@
+/******************************************************************************
+ *
+ * Project:  PROJ
+ * Purpose:  Test pj_phi2.c
```
Comment (mloskot) [inline test/unit/pj_phi2_test.cpp]: I think this should read "Test pj_phi2 function".

File: test/unit/CMakeLists.txt [hunk 3]
```diff
@@ -57,3 +57,11 @@ target_link_libraries(proj_test_unit
   gtest
   ${PROJ_LIBRARIES})
 add_test(NAME proj_test_unit COMMAND proj_test_unit)
+
+add_executable(pj_phi2_test
```
Comment (rouault) [inline test/unit/CMakeLists.txt:61]: I'm wondering if we need to create a dedicated test program for that ? Seems a bit overkill. We could put that as an extra source of the existing test. Same for automake

Comment (schwehr) [reply test/unit/CMakeLists.txt:61]: I think we should start with separate targets.  Keep test files as isolated from each other as possible.  A test could accidentally mess up errno, the default ctx, default logging or other state.  In fact tests should be in there that do that kind of thing.

File: test/unit/pj_phi2_test.cpp [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    EXPECT_DOUBLE_EQ(M_PI_2, pj_phi2(ctx, 0.0, -1.0));
+
+    EXPECT_DOUBLE_EQ(0.0, pj_phi2(ctx, 1.0, 1.0));
+    EXPECT_DOUBLE_EQ(M_PI, pj_phi2(ctx, -1.0, -1.0));
+
+    // TODO(schwehr): M_PI_4, M_PI_2, M_PI, M_E
+    // https://www.gnu.org/software/libc/manual/html_node/Mathematical-Constants.html
+
+    EXPECT_DOUBLE_EQ(-0.95445818456292697, pj_phi2(ctx, M_PI, 0.0));
+    EXPECT_TRUE(isnan(pj_phi2(ctx, 0.0, M_PI)));
```
Comment (rouault) [inline test/unit/pj_phi2_test.cpp]: should be std::isnan() to please x86_64-w64-mingw32

**Comment by schwehr** [comment]
Fixing... https://travis-ci.com/OSGeo/proj.4/jobs/128419573
```
../../../test/unit/pj_phi2_test.cpp:82:47: error: ‘isnan’ was not declared in this scope
     EXPECT_TRUE(isnan(pj_phi2(ctx, -inf, -inf)));
```

**Comment by rouault** [comment]
The MSVC cmake builds with ```pj_phi2_test.obj : error LNK2019: unresolved external symbol pj_phi2 referenced in function```
The issue is that pj_phi2() isn't exported in proj.def. So perhaps this test should be linked against the static version of the library (hoping that the symbols would then be visible) ? Or if that's too annoying to do, skip that test in the MSVC cmake case ?

**Comment by schwehr** [comment]
@rouault There are going to be a lot of tests like this.  I don't want to be changing what symbols are exported for externally facing builds.  How do I change the MSVC build to be static?

Anyone else have other ideas on how to handle this cleanly?

**Comment by rouault** [comment]
In appveyor.yml, we do -DBUILD_LIBPROJ_SHARED=ON and looking at src/lib_proj.cmake, I've the feeling that it is either a shared or a static lib build, but not both at the same time. We do want appveyor to test DLL builds. What we could potentially do is have appveyor do 2 builds: one shared one, without that test, and on static one with that test.

**Comment by rouault** [comment]
I've implemented my above proposal in https://github.com/rouault/proj.4/commit/6b0c94e47962164f20d858bda344f6fa6aeaf625 . Results in successful builds : https://ci.appveyor.com/project/rouault/proj-4/build/1.0.110 and https://travis-ci.org/rouault/proj.4/builds/389972491

File: test/unit/pj_phi2_test.cpp [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+    EXPECT_TRUE(std::isnan(pj_phi2(ctx, 0.0, inf)));
+
+    EXPECT_DOUBLE_EQ(4.7123889803846897, pj_phi2(ctx, -inf, 0.0));
+    EXPECT_TRUE(std::isnan(pj_phi2(ctx, 0.0, -inf)));
+
+    EXPECT_TRUE(std::isnan(pj_phi2(ctx, inf, inf)));
+    EXPECT_TRUE(std::isnan(pj_phi2(ctx, -inf, -inf)));
+}
+
+}  // namespace
```
Comment (kbevers) [inline test/unit/pj_phi2_test.cpp:85]: weird line ending here...

**Comment by mloskot** [comment]
I may be missing something, but AppVeyor already sets common `CMAKE_RUNTIME_OUTPUT_DIRECTORY_RELEASE` for EXEs and DLLs, so running the shared build should work fine, I'd expect.

Typically, within single build directory, it is either shared or static. 

BTW, CMake offers [BUILD_SHARED_LIBS](https://cmake.org/cmake/help/latest/variable/BUILD_SHARED_LIBS.html). There should be no need for custom ` BUILD_LIBPROJ_SHARED` option.

**Comment by kbevers** [comment]
Added in #1045. Closing this one.

**Pull request closed without merging**